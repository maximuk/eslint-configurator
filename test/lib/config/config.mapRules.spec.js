const { BASE_CONFIG_NAME } = require('../../../lib/const.js');
const Config = require('../../../lib/config.js');

describe('./lib/config.js :: mapRules', () => {
  it('should return modified rules', () => {
    const self = { name: BASE_CONFIG_NAME, prefix: 'prefix/' };
    const plugin = { rules: {
      testRule1: {
        meta: {
          deprecated: true,
          docs: {
            category: 'category',
            replacedBy: ['replacedBy'],
          },
          schema: ['schema'],
        },
      },
      testRule2: {
        meta: {
          schema: { key: 'value' },
        },
      },
      testRule3: {},
    } };
    const result = Config.prototype.mapRules.call(self, plugin);

    expect(result).toMatchSnapshot();
  });
});
