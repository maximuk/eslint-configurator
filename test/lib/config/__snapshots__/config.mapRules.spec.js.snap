// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./lib/config.js :: mapRules should return modified rules 1`] = `
Object {
  "prefix/testRule1": Object {
    "category": "category",
    "deprecated": true,
    "description": undefined,
    "link": "ðŸ”§ https://eslint.org/docs/rules/testRule1",
    "name": "prefix/testRule1",
    "replacedBy": "replacedBy",
  },
  "prefix/testRule2": Object {
    "category": "other",
    "deprecated": undefined,
    "description": undefined,
    "link": "ðŸ”§ https://eslint.org/docs/rules/testRule2",
    "name": "prefix/testRule2",
    "replacedBy": undefined,
  },
  "prefix/testRule3": Object {
    "category": "other",
    "deprecated": undefined,
    "description": undefined,
    "link": "https://eslint.org/docs/rules/testRule3",
    "name": "prefix/testRule3",
    "replacedBy": undefined,
  },
}
`;
